muxrpc/handlers/tunnel/server/state.go

```diff
@@ -5,30 +5,74 @@ package server
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"sync"
 	"time"
 
 	"github.com/ssb-ngi-pointer/go-ssb-room/internal/network"
 	"github.com/ssb-ngi-pointer/go-ssb-room/roomdb"
 	"github.com/ssb-ngi-pointer/go-ssb-room/roomstate"
-	refs "go.mindeco.de/ssb-refs"
 
 	kitlog "github.com/go-kit/kit/log"
 	"go.cryptoscope.co/muxrpc/v2"
 )
 
 type Handler struct {
 	logger kitlog.Logger
-	self   refs.FeedRef
 
+	netInfo network.ServerEndpointDetails
 	state   *roomstate.Manager
 	members roomdb.MembersService
 	config  roomdb.RoomConfig
 }
 
-func (h *Handler) isRoom(context.Context, *muxrpc.Request) (interface{}, error) {
-	return true, nil
+type IsRoomReply struct {
+	Name       string   `json:"name"`
+	Membership bool     `json:"membership"`
+	Features   []string `json:"features"`
+}
+
+func (h *Handler) isRoom(ctx context.Context, req *muxrpc.Request) (interface{}, error) {
+	ref, err := network.GetFeedRefFromAddr(req.RemoteAddr())
+	if err != nil {
+		return nil, err
+	}
+
+	pm, err := h.config.GetPrivacyMode(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var reply IsRoomReply
+	reply.Name = h.netInfo.Domain
+
+	// check if caller is a member
+	if _, err := h.members.GetByFeed(ctx, *ref); err != nil {
+		if errors.Is(err, roomdb.ErrNotFound) {
+			reply.Membership = false
+		}
+		return nil, err
```

Comment:
Don't we need an `else` here?

Reply:
It was a bit convoluted, i think and hope the new version is clearer.

Reply:
looks good!